### PR TITLE
Migrate to zlib-ng, part 3: Remove zlib and zlib-intel source code and license mentions (second attempt)

### PR DESCRIPTION
**Pull request by carlossanlop**
Contributes to: https://github.com/dotnet/runtime/issues/101465

Split in commits for easier review.

Second attempt of https://github.com/dotnet/runtime/pull/104399 but reverts the changes to `install-native-dependencies.sh` (775729ca1be8f726d25f31ac0a8992cdd2e75876).

The problem was that when running these two commands:
```
eng/install-native-dependencies.sh
./build.sh clr+libs
```

The error below would show up because cdacreader.csproj uses the LKG toolchain, which doesn't have the zlib-ng available. The workaround is to keep installing the system zlib on Unix. We can update `install-native-dependencies.sh` again once we consume a newer version of the SDK.

```
  cdac-build-tool -> /home/carlos/runtime/artifacts/bin/coreclr/linux.x64.Debug/cdac-build-tool/cdac-build-tool.dll
  /usr/bin/ld.bfd: cannot find -lz: No such file or directory
clang : error : linker command failed with exit code 1 (use -v to see invocation) [/home/carlos/runtime/src/native/managed/cdacreader/src/cdacreader.csproj]
/home/carlos/.nuget/packages/microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/build/Microsoft.NETCore.Native.targets(368,5): error MSB3073: The command ""clang" "/home/carlos/runtime/artifacts/obj/cdacreader/Debug/net9.0/linux-x64/native/libcdacreader.o" -o "/home/carlos/runtime/artifacts/bin/cdacreader/Debug/net9.0/linux-x64/native/libcdacreader.so" -Wl,--version-script=/home/carlos/runtime/artifacts/obj/cdacreader/Debug/net9.0/linux-x64/native/libcdacreader.exports -Wl,--export-dynamic -Wl,-soname=cdacreader.so -gz=zlib -fuse-ld=bfd /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libbootstrapperdll.o /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libRuntime.WorkstationGC.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libeventpipe-disabled.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libRuntime.VxsortEnabled.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libstandalonegc-disabled.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libstdc++compat.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/framework/libSystem.Native.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/framework/libSystem.IO.Compression.Native.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/framework/libSystem.Net.Security.Native.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/framework/libSystem.Security.Cryptography.Native.OpenSsl.a -g -Wl,-rpath,'$ORIGIN' -Wl,--build-id=sha1 -Wl,--as-needed -Wl,-e0x0 -pthread -ldl -lz -lrt -lm -shared -Wl,-z,relro -Wl,-z,now -Wl,--eh-frame-hdr -Wl,--discard-all -Wl,--gc-sections" exited with code 1. [/home/carlos/runtime/src/native/managed/cdacreader/src/cdacreader.csproj]
```

I opened an issue to track the removal of those 3 packages from the script: https://github.com/dotnet/runtime/issues/105373

### CODE REVIEW

File: docs/workflow/requirements/linux-requirements.md [hunk 1]
```diff
@@ -47,15 +47,14 @@ Install the following packages for the toolchain:
 * liblttng-ust-dev
 * libssl-dev
 * libkrb5-dev
-* zlib1g-dev
```
Comment (jkotas) [inline docs/workflow/requirements/linux-requirements.md:50]: If somebody follows these instructions to setup their environment, they will again get the build break about missing system zlib.

I think you want to split this PR into two: Deleting zlib zlib-intel sources that can be merged now, and deleting the system zlib from the dev environment that can merged only once we update the SDK used to build to P7.

Comment (carlossanlop) [reply docs/workflow/requirements/linux-requirements.md]: Removed here.

**Review by jkotas — APPROVED** [review]
Thanks

File: docs/workflow/requirements/linux-requirements.md (hunk 1 above)
Comment (carlossanlop) [reply docs/workflow/requirements/linux-requirements.md]: Here's the follow-up PR: https://github.com/dotnet/runtime/pull/105419

**Comment by carlossanlop** [comment]
All CI failures are known.

**Pull request merged**